### DISCUSSION TITLE
Ill-behaved equation of state calculation for FCC Al

### DISCUSSION THREAD

**Discussion by eric-yu-zhu** [CederGroupHub/chgnet#137] [Q&A] [answered]
Hi all,

I ran into a problem in a bulk modulus calculation using chgnet. The system is fcc Al, and the energy as a function of unit cell volume exhibits an abrupt change (see blue dots in the plot below). The problem can be easily reproduced with the script below. The same script worked well for bulk Cu and MOF materials.

Do you have any explanation or suggestion?

Thanks,
Eric

![eos_fccAl_chgnet](https://github.com/CederGroupHub/chgnet/assets/43016856/396c30b1-52d2-412c-ab49-736057a0b829)

```py
## import
import numpy as np
import ase
from calc_bulk_modulus import calc_bulk_modulus


## structure
from ase.build import bulk
atoms = bulk('Al', 'fcc', a=4.05)


## force model
from chgnet.model.model import CHGNet
from chgnet.model.dynamics import CHGNetCalculator
use_device = 'cpu'
chgnet = CHGNet.load()
dft_calc = CHGNetCalculator(model=chgnet, use_device=use_device)
atoms.calc = dft_calc


## parameters
eos_type = 'murnaghan'
scale_list = np.linspace(0.95, 1.15, 41)


## calc bulk modulus
eos, v0, e0, B = calc_bulk_modulus(atoms, scale_list, eos_type)    
B_GPa = B / ase.units.GPa
eos.plot()

```

**Comment by bowen-bd** [comment, marked as answer]
Using the same code, I get a continuous curve.
Are you using CHGNet v0.2.0?
If, so please update to the most recent version of CHGNet.
Current version defaults to v0.3.0 pretrained weights by `chgnet=CHGNet.load()`

![image](https://github.com/CederGroupHub/chgnet/assets/84425382/cd521b7f-1058-49fb-b6c4-219027a2c3d8)

**Reply by bowen-bd** [reply]
For cause of the v0.2.0 version bug, please refer to https://github.com/CederGroupHub/chgnet/issues/79
